MVCapplication/src/main/java/com/softserve/if072/mvcapp/service/UserService.java

```diff
@@ -47,7 +39,13 @@ public UserService(RestTemplate restTemplate, RegistrationService registrationSe
      * @return current User instance
      */
     public User getCurrentUser() {
-        User user = restTemplate.getForObject(getCurrentUserURL, User.class);
+        ServletRequestAttributes attr = (ServletRequestAttributes) RequestContextHolder.currentRequestAttributes();
+        HttpSession session = attr.getRequest().getSession();
+        User user = (User) session.getAttribute("user");
+        if (user == null) {
+            user = restTemplate.getForObject(getCurrentUserURL, User.class);
+            session.setAttribute("user", user);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Make "User" serializable or don't store it in the session. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS2441)